design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,231 @@
+# Stacks Security and Isolation
+
+* Owner: Jared Watts (@jbw976)
+* Reviewers: Crossplane Maintainers
+* Status: Draft, revision 1.0
+
+## Background
+
+The current model for Stacks, the means of extending Crossplane with new functionality, allows the stack author to declare arbitrary permissions that the stack will be granted by the Stack Manager.
+If the Stack author declares they need to update nodes or create volumes, those permissions will be granted.
+
+However, Crossplane is not an arbitrary Kubernetes cluster to run typical compute and storage workloads.
+Instead, it is a centralized control plane intended to manage resources external to the control plane (e.g. cloud provider services) and intelligently schedule compute workloads into remote clusters to consume these services.
+There is not really a need for Stacks being installed into the control plane to request arbitrary permissions.
+Crossplane can be more opinionated about the types of permissions granted within its control plane and lock things down in a more secure way.
+
+Futhermore, stacks are currently scoped to a single type that is used in all scenarios and by all personas.
+All new types of functionality, from an end user wanting to install a simple application to an administrator wanting to install support for low level infrastructure types, are all treated the same.
+They have the same approach to security and access control and the artifacts that they install have the same scope of impact to the control plane.
+This one size fits all approach is not ideal though, the multiple scenarios and personas need finer grain control around security, access, and isolation.
+
+This design doc proposes a new approach for the Crossplane Stack Manager (SM) that will support a smaller set of security permissions and increased control for the installation of stacks.
+
+## Design Summary
+
+The key differences in this design update are in the **scope** of the permissions that stacks can be granted within the control plane.
+
+* **Cluster Stacks** will be given permissions to watch, access, and create resources at a **cluster level**, i.e. across **all namespaces**.
+* **Stacks** will be given permissions only within a **single namespace**.
+* All stacks will be granted permissions to a small subset of core Kubernetes resources as well as access to any CRDs they own and depend on.
+
+This means that stacks should not be able to directly affect any resources outside of the specific namespace they are installed into, while cluster stacks operate at a higher level of privilege and can affect the entire cluster.
+This is a familiar concept to the Kubernetes permission scoping model of `ClusterRole` and `Role`.
+
+### CRD Scoping
+
+To avoid a common point of confusion, let's review the available scoping of CRDs.
+Custom resource **instances** can be scoped to either a single namespace or to the entire cluster.
```

Comment:
@jbw976 will we let [Cluster]Stacks create cluster-scoped CRDs?

Reply:
I'm adding some additional thoughts @bassam asked in Slack which are related to this question, but not entirely the same.

> One question that comes up for me is whether we will support cluster-scoped CRDs as a result of installing a stack (clusterstack or plain stack)
> in what scenarios will a Stack own a cluster-scoped CRD?